### ISSUE TITLE
Quick Search component: show results with partial match

### ISSUE THREAD

**Issue by joaotab** [Adobe-Marketing-Cloud/aem-core-wcm-components#149]
It would be nice if the Quick Search component would provide a way to return results with partial matches, e.g. if a page has the title "Trends" it would show up when typing "tren".

It works when we manually include a wildcard in the search ("tren*") but most users won't be aware of this behaviour

**Comment by gabrielwalt** [comment]
I agree that this would be required. (CQ-4247357)

**Comment by puradawid** [comment]
My very first proposition is to put that * into search expression automatically.

I am partially impacted by this problem because users are typing the name of the product but it cannot be found, due to additional special characters like (R).

Therefore, we will consider two cases:

- when a user types "sear" it looks for all "sear*"
- when there is a user that learnt the old-fashioned pattern and type "sear*" it searches for "sear**" which (probably) gives the same result

Additionally, I don't think adding an asterisk before the searched phrase is useful.

@joaotab, what is your opinion? I think I will prep some PR with this.

**Comment by joaotab** [comment]
I agree, this is actually what we did to workaround this limitation: we forked the SearchResultServlet and used a different selector.

Unfortunately this was a long time ago so our code is pretty outdated, but essentially it does what you've described: add a trailing '*'. It worked fine even if the user also typed the '*'.
